File: shared_resources/src/main/java/com/thewizrd/shared_resources/utils/CustomException.java
package com.thewizrd.shared_resources.utils;

import android.content.Context;

import androidx.annotation.Nullable;
import androidx.annotation.StringRes;

import com.thewizrd.shared_resources.SimpleLibrary;

public class CustomException extends Exception {
    @StringRes
    int stringResId = Integer.MIN_VALUE;

    public CustomException(@StringRes int stringResId) {
        this.stringResId = stringResId;
    }

    @Nullable
    @Override
    public String getMessage() {
        if (stringResId != Integer.MIN_VALUE) {
            Context context = SimpleLibrary.getInstance().getAppContext();
            return context.getString(stringResId);
        }

        return super.getMessage();
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/utils/CustomException.kt
package com.thewizrd.shared_resources.utils

import androidx.annotation.StringRes
import com.thewizrd.shared_resources.sharedDeps

class CustomException(@StringRes stringResId: Int) : Exception() {
    @StringRes
    var stringResId = Int.MIN_VALUE

    init {
        this.stringResId = stringResId
    }

    override val message: String?
        get() {
            if (stringResId != Int.MIN_VALUE) {
                return sharedDeps.context.getString(stringResId)
            }

            return super.message
        }
}